helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java

```diff
@@ -0,0 +1,480 @@
+
+  private static final Logger log = LoggerFactory.getLogger(HeliosSoloDeployment.class);
+
+  public static final String BOOT2DOCKER_SIGNATURE = "Boot2Docker";
+  public static final String PROBE_IMAGE = "onescience/alpine:latest";
+  public static final String HELIOS_IMAGE = "spotify/helios-solo:latest";
+  public static final String HELIOS_NAME_PREFIX = "solo.local.";
+  public static final String HELIOS_CONTAINER_PREFIX = "helios-solo-container-";
+  public static final int HELIOS_MASTER_PORT = 5801;
+
+  final DockerClient dockerClient;
+  final DockerHost dockerHost;
+  final DockerHost containerDockerHost;
+  final String namespace;
+  final List<String> env;
+  final List<String> binds;
+  final String heliosContainerId;
+  final HeliosClient heliosClient;
+
+  HeliosSoloDeployment(final Builder builder) {
+    final String username = Optional.fromNullable(builder.heliosUsername).or(randomString());
+
+    this.dockerClient = checkNotNull(builder.dockerClient, "dockerClient");
+    this.dockerHost = Optional.fromNullable(builder.dockerHost).or(DockerHost.fromEnv());
+    this.containerDockerHost = Optional.fromNullable(builder.containerDockerHost)
+            .or(containerDockerHostFromEnv());
+    this.namespace = Optional.fromNullable(builder.namespace).or(randomString());
+    this.env = containerEnv();
+    this.binds = containerBinds();
+
+    final String heliosHost;
+    final String heliosPort;
+    //TODO(negz): Determine and propagate NetworkManager DNS servers?
+    try {
+      assertDockerReachableFromContainer();
+      if (dockerHost.address().equals("localhost") || dockerHost.address().equals("127.0.0.1")) {
+          heliosHost = containerGateway();
+      } else {
+        heliosHost = dockerHost.address();
+      }
+      this.heliosContainerId = deploySolo(heliosHost);
+      heliosPort = getHostPort(this.heliosContainerId, HELIOS_MASTER_PORT);
+    } catch (HeliosDeploymentException e) {
+      throw new AssertionError("Unable to deploy helios-solo container.", e);
+    }
+
+    // Running the String host:port through HostAndPort does some validation for us.
+    this.heliosClient = HeliosClient.newBuilder()
+            .setUser(username)
+            .setEndpoints("http://" +
+                    HostAndPort.fromString(dockerHost.address() + ":" + heliosPort))
+            .build();
+  }
+
+  private DockerHost containerDockerHostFromEnv() {
+    if (isBoot2Docker(dockerInfo())) {
+      return DockerHost.from(DefaultDockerClient.DEFAULT_UNIX_ENDPOINT, null);
+    } else {
+      return DockerHost.fromEnv();
+    }
+  }
+
+  private Boolean isBoot2Docker(final Info dockerInfo) {
+    return dockerInfo.operatingSystem().contains(BOOT2DOCKER_SIGNATURE);
+  }
+
+  private Info dockerInfo() {
+    try {
+      return this.dockerClient.info();
+    } catch (DockerException | InterruptedException e) {
+      // There's not a lot we can do if Docker is unreachable.
+      throw new AssertionError(e);
```

Comment:
`throw RuntimeException(e)` or `throw Throwables.propogate(e)` would be more idiomatic